src/bot.ts

```diff
@@ -106,14 +99,61 @@ Also, here are some links you might find useful:
   )
 }
 
-function _handleFetchDataError (error: Error): string {
-  log.error(error)
-  return 'N/A'
-}
-
-dfusionRepo.watchOrderPlacement({
-  onNewOrder (event) {
-    bot.sendMessage(channelId, formatNewOrders(event))
+dfusionService.watchOrderPlacement({
+  onNewOrder (order) {
+    const {
+      // owner,
+      buyToken,
+      sellToken,
+      validFrom,
+      validUntil,
+      // validFromBatchId,
+      // validUntilBatchId,
+      priceNumerator,
+      priceDenominator
+      // event
+    } = order
+
+    // Calculate the price
+    let price
```

Comment:
```js
const precisionFactor = 10 ** Math.abs(buyToken.decimals - sellToken.decimals)
const price = priceNumerator.multipliedBy(precisionFactor).dividedBy(priceDenominator)
```
?

Reply:
It's a good suggestion and it seems it would work, but it wouldn't work exactly like that.

Note that depending on the token with more precision, we add 0s to the other (in the numerator or denominator depending on the case)

This will be made an util shortly in dex-js